plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/DecimalCoercers.java

```diff
@@ -228,6 +235,61 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
         }
     }
 
+    public static Function<Block, Block> createDecimalToVarcharCoercer(DecimalType fromType, VarcharType toType)
+    {
+        int lengthLimit = toType.getLength().orElse(VarcharType.UNBOUNDED_LENGTH);
+        if (fromType.isShort()) {
+            return new ShortDecimalToVarcharCoercer(fromType, lengthLimit);
+        }
+        return new LongDecimalToVarcharCoercer(fromType, lengthLimit);
+    }
+
+    private static class ShortDecimalToVarcharCoercer
+            extends TypeCoercer<DecimalType, VarcharType>
+    {
+        private final int lengthLimit;
+
+        protected ShortDecimalToVarcharCoercer(DecimalType fromType, int lengthLimit)
+        {
+            super(fromType, VARCHAR);
+            this.lengthLimit = lengthLimit;
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            String stringValue = Decimals.toString(fromType.getLong(block, position), fromType.getScale());
+            // Hive truncates digits before the decimal point, which can be perceived as a bug
```

Comment:
```suggestion
            // Hive truncates digits (also before the decimal point), which can be perceived as a bug
```

Reply:
hmm I wanted to emphasise that truncation BEFORE decimal point can be perceived as bug not truncation in general. But ok

Reply:
I see your point.
I want, however, to avoid expressing that truncating AFTER decimal point is never perceived as a bug. Or that Hive does not do that at all.